src/Microsoft.AspNetCore.Cors/Infrastructure/CorsService.cs

```diff
@@ -245,6 +245,11 @@ private void AddOriginToResult(string origin, CorsPolicy policy, CorsResult resu
             else if (policy.Origins.Contains(origin))
             {
                 result.AllowedOrigin = origin;
+
+                if(policy.Origins.Count > 1)
```

Comment:
@kichalla @Eilon we may need to discuss this and possibly change this to `policy.Origins.Count >= 1`. The CORS spec does not mention the header being set for a single origin but it looks like that's what Chrome expects for caching to correctly work as @andrew-laughlin explained.